### PR TITLE
all655: Updated Scripts, ReadMe, and TOC

### PR DESCRIPTION
**Pull request by all655**
Hello Dr. Jones,

I updated all our install scripts to be more efficient and reliable. I have also documented and tested them well.

In documenting, the directories get messed up in the script, which I promptly fixed. They should still work well with our existing CI/CD.

Additionally, the ReadMe and TOC have been updated to reflect these changes.

Thank you,
Lael Lum

### CODE REVIEW

**Comment by all655** [comment]
Note: 
One of the commits says: "Latest changes for Dr. Jones to review in first pull request." Unfortunately that is no longer true as I discovered some directory issues that needed correcting.

File: scripts/install-gnb.sh [hunk 1]
```diff
@@ -29,23 +31,11 @@ Group=nogroup
 [Install]
 WantedBy=multi-user.target
 EOF
-
-# Install build tools
+# \
```
Comment (bjones1) [inline scripts/install-gnb.sh:34]: Why the `\`?

File: scripts/install-open5gs.sh [hunk 2]
```diff
@@ -1,5 +1,7 @@
 #!/bin/bash
-
+#   
+# # install-open5gs.sh
```
Comment (bjones1) [inline scripts/install-open5gs.sh:3]: Use ``# # `install-open5gs.sh` -- a script to....``.

File: scripts/install-open5gs.sh [hunk 3]
```diff
@@ -8,14 +10,19 @@ fi
...
 PS4='[DEBUG] '
 set -x
-
+# \
+# **Install Docker and needed libraries:**
+sudo apt-get install -y cmake make gcc g++ pkg-config libfftw3-dev libmbedtls-dev libsctp-dev libyaml-cpp-dev libgtest-dev
+sudo apt install docker-compose	
+sudo apt install docker.io	
+# \
 # Add Docker's official GPG key:
```
Comment (bjones1) [inline scripts/install-open5gs.sh:19]: Does this script come from elsewhere? Guessing https://docs.docker.com/engine/install/ubuntu/. If so, cite.

File: scripts/install-ue.sh [hunk 4]
```diff
@@ -2,42 +2,57 @@
...
 
 PS4='[DEBUG] '
 set -x
+# \
+# **Install Docker and needed libraries:**
+sudo apt-get install -y cmake make gcc g++ pkg-config libfftw3-dev libmbedtls-dev libsctp-dev libyaml-cpp-dev libgtest-dev 
+sudo apt install docker-compose	
+sudo apt install docker.io
+# \
+# Add Docker's official GPG key:
```
Comment (bjones1) [inline scripts/install-ue.sh:20]: Cite official install script.

File: scripts/install-ue.sh [hunk 5]
```diff
@@ -2,42 +2,57 @@
...
 
-# Install build tools
-#
-# Install the required build tools for the
-# [srsRAN 4G](https://docs.srsran.com/projects/4g/en/latest/general/source/1_installation.html#installation-from-source)
-# and the
-# [srsRAN project](https://docs.srsran.com/projects/project/en/latest/user_manuals/source/installation.html#manual-installation).
+# \
+# Install build tools:
 apt-get update && apt-get upgrade -y
```
Comment (bjones1) [inline scripts/install-ue.sh:37]: Why the removed docs? If instructions here came from elsewhere, cite them.

**Review by bjones1 — CHANGES_REQUESTED** [review]

File: scripts/install-gnb.sh (hunk 1 above)
Comment (all655) [reply scripts/install-gnb.sh:14]: It looked better that way in CodeChat. I can remove it.

**Comment by all655** [comment]
Hello Dr. @bjones1 , 

Ready for another review.

Thank you,
Lael Lum

File: README.md [hunk 6]
```diff
@@ -4,41 +4,78 @@ Penetration Testing Tool for Software Defined Radio
...
+
+**To run the gNB run (Machine B):**
+
+```bash
+sudo gnb -c ./soft-t-ue/configs/zmq/gnb_zmq.yaml
 ```
 
 ## System Architecture
+
 ![Soft-T-UE-System.png](https://github.com/oran-testing/soft-t-ue/blob/grafana_integration/docs/images/Soft-T-UE-System.png)
```
Comment (bjones1) [inline README.md:81]: Wrong URL -- try ``. (But the diagram is too large and should be shrunk some.)

File: scripts/install-open5gs.sh [hunk 7]
```diff
@@ -8,14 +10,20 @@ fi
 
 PS4='[DEBUG] '
 set -x
-
+# \
+# [Docker](https://docs.docker.com/engine/install/ubuntu/)
```
Comment (bjones1) [inline scripts/install-open5gs.sh:14]: Add a newline after Docker?

File: scripts/install-open5gs.sh [hunk 8]
```diff
@@ -24,13 +32,19 @@ echo \
...
 cd srsRAN_Project
-git checkout ue-tester
+mkdir build
+cd build
+cmake ../ -DENABLE_EXPORT=ON -DENABLE_ZEROMQ=ON
+make -j $(nproc)
+sudo make install
+# Sometimes Docker misbehaves and can be fixed with a simple restart:
+sudo systemctl restart docker
+# The status of Docker can be checked at anytime with: sudo systemctl status
```
Comment (bjones1) [inline scripts/install-open5gs.sh:43]: Put in a code font: `sudo systemctl status`

File: scripts/install-open5gs.sh [hunk 9]
```diff
@@ -24,13 +32,19 @@ echo \
...
+# docker
+cd ..
+# If desired, the UE can be tested with: git checkout ue-tester
```
Comment (bjones1) [inline scripts/install-open5gs.sh:46]: Code font also.

File: scripts/install-ue.sh [hunk 10]
```diff
@@ -2,16 +2,38 @@
...
+# Check if the script is run as root:
 if [ "$EUID" -ne 0 ]; then
 	echo "This script must be run as root."
 	exit 1
 fi
 
 PS4='[DEBUG] '
 set -x
+# \
+# [Docker](https://docs.docker.com/engine/install/ubuntu/)
```
Comment (bjones1) [inline scripts/install-ue.sh:15]: Missing newline

File: scripts/install-ue.sh [hunk 11]
```diff
@@ -23,21 +45,22 @@ apt-get install -y cmake make gcc g++ pkg-config libfftw3-dev libmbedtls-dev lib
...
 cd soft-t-ue
 mkdir -p build
 cd build
-cmake ../
+cmake ../ -DENABLE_EXPORT=ON -DENABLE_ZEROMQ=ON
 make -j $(nproc)
 make install
+# Sometimes Docker misbehaves and can be fixed with a simple restart:
+sudo systemctl restart docker
+# The status of Docker can be checked at anytime with: sudo systemctl status docker
```
Comment (bjones1) [inline scripts/install-ue.sh:60]: Code font

File: toc.md [hunk 12]
```diff
@@ -1,9 +1,10 @@
 # The T-UE project
 
-- [Introduction](README.md)
+- Introduction
+  - [ReadMe](README.md)
 - Install scripts
+  - [Install gNB](scripts/install-gnb.sh)
+  - [Install 5gs](scripts/install-open5gs.sh)
```
Comment (bjones1) [inline toc.md:7]: Probably Install open5gs

**Review by bjones1 — DISMISSED** [review]

File: README.md (hunk 6 above)
Comment (all655) [reply README.md]: I did not resize the image as it behaved much better in the README.md as it was. Leaving it as is allowed it to scale and not squish/compress when resizing the window. However, in CodeChat, it was too large.

File: scripts/install-open5gs.sh (hunk 7 above)
Comment (all655) [reply scripts/install-open5gs.sh]: For some reason in CodeChat the linked references do not seem work. I would like to discuss them with you in our next meeting.

**Review by cueltschey — APPROVED** [review]

**Pull request merged**